c/polyjuice_utils.h

```diff
@@ -38,28 +32,58 @@ void debug_print_data(const char *prefix, const uint8_t *data,
   g_debug_buffer[offset] = '\0';
   ckb_debug(g_debug_buffer);
 }
-void debug_print_int(const char* prefix, int64_t ret) {
+void debug_print_int(const char *prefix, int64_t ret) {
   sprintf(g_debug_buffer, "%s => %ld", prefix, ret);
   ckb_debug(g_debug_buffer);
 }
-#endif /* NO_DEBUG_LOG */
+// avoid VM(InvalidEcall(80))
+int printf(const char *format, ...) { return 0; }
+#else
+#undef ckb_debug
+#define ckb_debug(s) \
+  do {               \
+  } while (0)
+#define debug_print(s) \
+  do {                 \
+  } while (0)
+#define debug_print_int(prefix, value) \
+  do {                                 \
+  } while (0)
+#define debug_print_data(prefix, data, data_len) \
+  do {                                           \
+  } while (0)
+int printf(const char *format, ...) { return 0; }
+#endif /* POLYJUICE_DEBUG_LOG */
 
 #define memset(dest, c, n) _smt_fast_memset(dest, c, n)
 
-/* polyjuice contract account (normal/create2) script args size*/
-static const uint32_t CONTRACT_ACCOUNT_SCRIPT_ARGS_SIZE = 32 + 4 + 20;
+/* https://stackoverflow.com/a/1545079 */
+#pragma push_macro("errno")
+#undef errno
+bool is_errno_ok(mol_seg_res_t *script_res) {
+  return script_res->errno == MOL_OK;
+}
+#pragma pop_macro("errno")
+
+gw_reg_addr_t init_reg_addr(const uint8_t eth_addr[ETH_ADDRESS_LEN]) {
```

Comment:
If return the `gw_reg_addr_t` from current stack memory, better use `new_reg_addr` than `init_reg_addr` as function name.

Reply:
https://github.com/nervosnetwork/godwoken-polyjuice/pull/138/commits/0ea8e4e4b8ded3dfb992293af381f3699f495900